admin_sso/auth.py

```diff
@@ -1,9 +1,49 @@
 from django.contrib.auth import get_user_model
-
 from admin_sso.models import Assignment
 
 
 class DjangoSSOAuthBackend(object):
+
+    def _get_username(self, sso_email):
```

Comment:
@matthiask How about this solution? You can overwrite this particular method and adjust things in the way you expect it in the field that's set as unique in your custom user model.

Based on https://docs.djangoproject.com/en/1.10/topics/auth/customizing/#django.contrib.auth.models.CustomUser.USERNAME_FIELD